packages/client/src/components/settings-dialog.vue

```diff
@@ -101,14 +96,36 @@ const { dialogRef, onDialogCancel, onDialogOK, onDialogHide } =
 
 const { t } = useI18n();
 
-const newSettings = reactive<SettingsUpdateInput>({
+const newSettings = reactive<RetailLocationSettingsFragment>({
   maxBookingDays: props.maxBookingDays,
   payOffEnabled: props.payOffEnabled,
   warehouseMaxBlockSize: props.warehouseMaxBlockSize,
+  buyRate: props.buyRate,
+  sellRate: props.sellRate,
 });
 
-const purchaseRate = props.buyRate;
-const saleRate = props.sellRate;
+function validateSettings() {
+  const settingsUpdate = {
+    type: "save" as const,
+    settings: newSettings,
+  };
+
+  if (
+    newSettings.buyRate !== props.buyRate ||
+    newSettings.sellRate !== props.sellRate
+  ) {
+    Dialog.create({
+      title: t("general.settings.updateRatesConfirmTitle"),
+      message: t("general.settings.updateRatesConfirmMessage"),
+      ok: t("actions.continue"),
+      cancel: t("common.cancel"),
+    }).onOk(() => {
+      onDialogOK(settingsUpdate);
```

Comment:
Seems like when this is invoked and then the dialog is re-opened, the data is not correctly updated to the new settings, because the data shown here is the same old one.
Probably just need to update the GraphQL cache here.